services/sourceforge/sourceforge-language.service.js

```diff
@@ -0,0 +1,45 @@
+import Joi from 'joi'
+import BaseSourceForgeService from './sourceforge-base.js'
+
+const schema = Joi.object({
+  categories: Joi.object({
+    language: Joi.array().required(),
+  }).required(),
+}).required()
+
+export default class SourceforgeLanguage extends BaseSourceForgeService {
+  static category = 'analysis'
+
+  static route = {
+    base: 'sourceforge/language',
+    pattern: ':index/:project',
+  }
+
+  static examples = [
+    {
+      title: 'SourceForge language',
+      namedParams: {
+        project: 'mingw',
+        index: '0',
+      },
+      staticPreview: this.render('Fortran'),
+    },
+  ]
+
+  static defaultBadgeData = { label: 'language' }
+
+  static render(language) {
+    return {
+      message: language,
+      color: 'blue',
+    }
+  }
+
+  async handle({ project, index }) {
+    const body = await this.fetch({ project, schema })
+    return this.constructor.render(
+      body.categories.language[parseInt(index)]?.fullname ||
```

Comment:
If we rely on the `fullname` property being present, we need to validate it exists in the schema
Here's an example of how to validate an array is an array of objects containing a given property
https://github.com/badges/shields/blob/1543d0f363f8d00dca27bac486b8b3cb112dbd81/services/pypi/pypi-base.js#L12-L18

---

Also, some of these names are a bit too wordy for a badge. For example "GNU General Public License version 2.0 (GPLv2)".

Do SourceForge publish anywhere a list of their identifiers? On the PyPI badge, we manually aliased some of the longer descriptions to SPDX identifiers https://github.com/badges/shields/blob/1543d0f363f8d00dca27bac486b8b3cb112dbd81/services/pypi/pypi-helpers.js#L69-L71 We might need to do the same here.